test/e2e/app-dir/app-invalid-revalidate/app-invalid-revalidate.test.ts

```diff
@@ -0,0 +1,78 @@
+import { createNextDescribe } from 'e2e-utils'
+import { check } from 'next-test-utils'
+
+createNextDescribe(
+  'app-invalid-revalidate',
+  {
+    files: __dirname,
+    skipStart: true,
+    skipDeployment: true,
+  },
+  ({ next }) => {
+    it('should error properly for invalid revalidate at layout', async () => {
+      await next.stop().catch(() => {})
+      const origText = await next.readFile('app/layout.tsx')
+
+      try {
+        await next.patchFile(
+          'app/layout.tsx',
+          origText.replace('// export', 'export')
+        )
+        await next.start().catch(() => {})
+
+        await check(async () => {
+          if ((global as any).isNextDev) {
+            await next.fetch('/')
+          }
+          return next.cliOutput
+        }, /Invalid revalidate value "1" on "\/", must be a non-negative number or "false"/)
+      } finally {
+        await next.patchFile('app/layout.tsx', origText)
+      }
+    })
+
+    it('should error properly for invalid revalidate at page', async () => {
+      await next.stop().catch(() => {})
+      const origText = await next.readFile('app/page.tsx')
+
+      try {
+        await next.patchFile(
+          'app/page.tsx',
+          origText.replace('// export', 'export')
+        )
+        await next.start().catch(() => {})
+
+        await check(async () => {
+          if ((global as any).isNextDev) {
+            await next.fetch('/')
+          }
```

Comment:
we could grab these flags from `createNextDescribe` right?

Reply:
Ah yeah updated, was using previous pattern